tests/unittests/tests-hashes/tests-hashes-keccak.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2017 Adrian Herrmann <adrian.herrmann@fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     unittests
+ * @{
+ *
+ * @file
+ * @brief       Test cases for the Keccak hashing package
+ *
+ * @author      Adrian Herrmann <adrian.herrmann@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "KeccakHash800.h"
+#include "embUnit.h"
+
+#define KeccakP200_excluded
+#define KeccakP400_excluded
+#define KeccakP1600_excluded
+
+Keccak_HashInstance* hashInstance;
+
+const unsigned int rate = 544;
+const unsigned int hashbitlen = 256;
+const unsigned char delimitedSuffix = '\1';
+
+const BitLength databitlen = 80;
+const char datastring[] = "0123456789";
+const BitSequence* data = (unsigned char*) datastring;
+
+const char hashvalstring[] =
+            "4c0ff0d8a1479dfe58fb0b55c7870e6c64e1c770fcd0bcb9fdc78d2570fb94a2";
+BitSequence* hashval;
+
+static void setUp(void)
+{
+    /* Initialize */
+    hashInstance = malloc(sizeof(Keccak_HashInstance));
+    hashval = malloc(hashbitlen / 8);
+}
+
+static void tearDown(void)
+{
+    /* Finalize */
+    free(hashInstance);
+    free(hashval);
+}
+
+static void test_keccak(void)
+{
+    /* Testing hash initialization */
+    HashReturn return_init = Keccak_HashInitialize(hashInstance, rate, 800-rate,
```

Comment:
Here, I would rather have a wrapper function like KeccakP800_Hash-256 or something like that.
The idea would be to: 
- abstract away some parameters so that it looks as much as possible like hash functions we already support
- make it clear in the name what we are using, so that users are not confused.

Reply:
An appropriate place for this wrapper function may be KeccakHash800.h inside the Modes directory of KeccakCodePackage, what do you think?

Reply:
My gut feeling is that this should be outside of the Keccak package.

Reply:
Forgive me if I'm misunderstanding your reasoning, but the way I see it I don't think it makes much sense for a function that wraps functions from a package to be outside said package, unless you would consider it outside of the package if it's part of its patch files.

Reply:
I've discussed with @miri64 today and she has agreed with you. However, I'm still a bit confused, so please let us discuss the following:

1. I make changes to the original KeccakCodePackage repository, namely to the Makefiles and also the inclusion of the KeccakHash800.c and KeccakHash800.h files. Should I include all the changes as patch files inside a subdirectory of the package's root directory (i.e. pkg/keccak) or should I handle this differently?
2. @miri64 mentioned that I should add the wrapper function in a subdirectory of pkg/keccak but not as a patch file and referred to a package she wrote herself - unfortunately, @miri64 had to leave and so did I when we discussed this so I don't really understand anymore how she meant it. Could you explain this to me? Where should I put the wrapper function?

Reply:
@adrianghc I also suggest to put the wrapper function in a subdirectory of the pkg.
@miri64 are you referring to a /contrib such as in [here](https://github.com/RIOT-OS/RIOT/tree/master/pkg/openthread/contrib) for instance?

Reply:
> @miri64 are you referring to a /contrib such as in here for instance?

Yes!

Reply:
So if I put the wrapper function inside KeccakHash800.c/KeccakHash800.h and then put those two files in a subdirectory of the pkg instead of a fork of the KeccakCodePackage repo, would that be fine?

In that case, how should I handle the changes to the Makefile.build, HighLevel.build and LowLevel.build files that I made in my fork of KeccakCodePackage so that I wouldn't need the fork and we could resort to the original repo for this package instead? Would the following work:

- Git patch files inside a subdirectory "patches" for all changed files (i.e. the *.build files listed above)
- A subdirectory "contrib" for all newly created files (i.e. the KeccakHash800 files and the CompilerFlags.build files)

Reply:
@adrianghc yes this approach seems good to me.

Reply:
This should be addressed now, let me know if it looks good.